pep-0710.rst

```diff
@@ -0,0 +1,418 @@
+
+The ``provenance_url.json`` JSON file MUST be a dictionary, compliant with
+:rfc:`8259` and UTF-8 encoded.
+
+If present, it MUST contain exactly two keys. The first one is ``url``, with
+type ``string``.  The second key MUST be ``archive_info`` with a value defined
+below.
+
+The value of the ``url`` key MUST be the URL from which the distribution package was downloaded. If a wheel is
+built from a source distribution, the ``url`` value MUST be the URL from which
+the source distribution was downloaded. If a wheel is downloaded and installed directly,
+the ``url`` field MUST be the URL from which the wheel was downloaded.
+As in the :ref:`direct URL origin specification<packaging:direct-url>`, the ``url`` value
+MUST be stripped of any sensitive authentication information for security reasons.
+
+The user:password section of the URL MAY however be composed of environment
+variables, matching the following regular expression:
+
+.. code-block:: text
+
+    \$\{[A-Za-z0-9-_]+\}(:\$\{[A-Za-z0-9-_]+\})?
+
+Additionally, the user:password section of the URL MAY be a well-known,
+non-security sensitive string. A typical example is ``git`` in the case of an
+URL such as ``ssh://git@gitlab.com``.
+
+The value of ``archive_info`` MUST be a dictionary with a single key
+``hashes``.  The value of ``hashes`` is a dictionary mapping hash function names to a
+hex-encoded digest of the file referenced by the ``url`` value. Multiple hashes
+can be included, and it is up to the consumer to decide what to do with
+multiple hashes (it may validate all of them or a subset of them, or nothing at
+all).
+
+Each hash MUST be one of the single argument hashes provided by
+:data:`py3.11:hashlib.algorithms_guaranteed`, excluding ``sha1`` and ``md5`` which MUST NOT be used.
+As of Python 3.11, with ``shake_128`` and ``shake_256`` excluded
+for being multi-argument, the allowed set of hashes is:
+
+.. code-block:: python
+
+  >>> import hashlib
+  >>> sorted(hashlib.algorithms_guaranteed - {"shake_128", "shake_256", "sha1", "md5"})
+  ['blake2b', 'blake2s', 'sha224', 'sha256', 'sha384', 'sha3_224', 'sha3_256', 'sha3_384', 'sha3_512', 'sha512']
+
+Each hash MUST be referenced by the canonical name of the hash, always lower case.
+
+Hashes ``sha1`` and ``md5`` MUST NOT be present, due to the security
+limitations of these hash algorithms. Conversely, hash ``sha256`` SHOULD
+be included.
+
+Installers that cache distribution packages from an index SHOULD keep
+information related to the cached distribution artifact, so that
+the ``provenance_url.json`` file can be created even when installing distribution packages
+from the installer's cache.
+
+Backwards Compatibility
+=======================
+
+Since this PEP specifies a new file in the ``*.dist-info`` directory, there are
+no backwards compatibility implications to consider in the ``provenance_url.json``
+file itself. Also, this proposal does not make any changes to the
```

Comment:
[Re-making this from before in revised form, since it got outdated anyway when this was moved]

It isn't quite true that there are no backward compat implications with adding this file, since installers are allowed to (and do, particularly with how license files are currently handled pre-PEP 639) add arbitrarily-named files to this directory. It's unlikely, but if any existing tool adds a file this this same name, it will conflict. For completeness, you might want to mention this, and any research/checks you've done to see if any existing tools might happen to be using a file with this name already.

Reply:
Okay, will include checks for this. Possible installers to be included: pip, Pipenv, poetry, pip-tools, installer, Conda, PDM. Please let me know if there is any missing or any should be excluded.

Reply:
Seems like this was (comprehensively) address, so resolving this and will make any followup-comments as necessary :+1: